### PR TITLE
Adding JSON log formatter

### LINKED ISSUE #196: Log plugin should use SerDe compatible format
For flexibility of querying from Athena we should use a SerDe compatible format:

https://docs.aws.amazon.com/athena/latest/ug/supported-format.html

Supported formats are CSV, Tab-separated values, JSON, Parquet (fast!!!), Logstash, Apache, and CloudTrail.

### PR DESCRIPTION
**Pull request by kolanos**
Closes #196

Signed-off-by: Michael Lavers <kolanos@gmail.com>

### CODE REVIEW

**Comment by pselle** [comment]
Requesting @ewindisch as the reviewer here, since she opened original issue

**Comment by coreylight** [comment]
I don't see a test that confirms the existence and format of the timestamp?
Also, could we perhaps use a key name more general like `datetime` or `timestamp`?

File: iopipe/contrib/logger/formatter.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+            record.message = record.message + record.exc_text
+        if hasattr(record, 'stack_info') and record.stack_info:
+            if record.message[-1:] != '\n':
+                record.message = record.message + '\n'
+            record.message = record.message + self.formatStack(record.stack_info)
+        return json.dumps({
+            'message': record.message,
+            'name': record.name,
+            'severity': record.levelname,
+            'time': record.asctime,
```
Comment (pselle) [inline iopipe/contrib/logger/formatter.py]: time => timestamp?

File: tests/contrib/logger/test_plugin.py [hunk 2]
```diff
@@ -28,13 +28,13 @@ def test__logger_plugin(mock_send_report, mock_get_signed_request, mock_upload_l
...
-    assert 'testlog - ERROR - And you have it, too.' in stream.getvalue()
-    assert "testlog - CRITICAL - And it's fatal." in stream.getvalue()
+    assert '"message": "I got nothing.", "name": "testlog", "severity": "DEBUG"' not in stream.getvalue()
+    assert '"message": "I might have something.", "name": "testlog", "severity": "INFO"' in stream.getvalue()
+    assert '"message": "Got something.", "name": "testlog", "severity": "WARNING"' in stream.getvalue()
+    assert '"message": "And you have it, too.", "name": "testlog", "severity": "ERROR"' in stream.getvalue()
+    assert '"message": "And it\'s fatal.", "name": "testlog", "severity": "CRITICAL"' in stream.getvalue()
 
-    assert 'testlog - INFO - This is not a misprint.' in stream.getvalue()
+    assert '"message": "This is not a misprint.", "name": "testlog", "severity": "INFO"' in stream.getvalue()
```
Comment (pselle) [inline tests/contrib/logger/test_plugin.py]: Shouldn't we look for the timestamp in here?

**Comment by pselle** [comment]
@kolanos oops, approved before seeing the test failure -- but code looking good

**Comment by kolanos** [comment]
@pselle Yeah, the failing test was a Python 2.7 compatibility issue with the logging formatter base class. Fixed now.

**Comment by pselle** [comment]
@kolanos Great. Yay tox!

**Pull request merged**